### PR TITLE
dry up increment/decrement

### PR DESCRIPTION
**Pull request by grosser**

### CODE REVIEW

File: activesupport/lib/active_support/cache/memory_store.rb [hunk 1]
```diff
@@ -167,6 +149,19 @@ def delete_entry(key, options) # :nodoc:
             !!entry
           end
         end
+
+        def modify_value(name, amount, options)
```
Comment (sgrif) [inline activesupport/lib/active_support/cache/memory_store.rb]: This should be private.

Comment (grosser) [inline activesupport/lib/active_support/cache/memory_store.rb]: done

File: activesupport/lib/active_support/cache/memory_store.rb [hunk 2]
```diff
@@ -167,6 +149,21 @@ def delete_entry(key, options) # :nodoc:
...
+          synchronize do
+            options = merged_options(options)
+            if num = read(name, options)
+              num = num.to_i + amount
+              write(name, num, options)
+              num
+            else
+              nil
```
Comment (sgrif) [inline activesupport/lib/active_support/cache/memory_store.rb]: We can drop the else clause entirely.

**Comment by sgrif** [comment]
Can you rebase as well? Master should be green now.

**Comment by grosser** [comment]
fixed and rebased

**Pull request merged**